src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,6 +88,16 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+    	try {
+    		File file = path.toFile();    	
+    		if (!file.exists() || file.isDirectory()){
+    			throw new FileNotFoundException();
+    		}
+    	}
+    	catch(FileNotFoundException fnfe){
+    		System.out.println("File not found, exiting");
```

Comment:
you have a UI class to talk to the user, why not use that instead of `sysout` here?